### ISSUE TITLE
Label Confidence Re-weighting

### ISSUE THREAD

**Issue by Janeey99** [wyxstriker/ReweightingDisfluency#1]
没有找到代码中关于这一部分的reweight操作，请问可以指一下具体在哪处代码体现的吗？

**Comment by wyxstriker** [comment]
非常抱歉，先前提交的代码是早期版本，没有包括label_confidence操作，目前已经更新，感谢指正。

实验中每轮的label confidence re-weighting通过权重文件`soft.tsv`传递，具体流程可以参考`src/teacher.py`：
- 教师模型打标阶段保留置信信息：
```python
# teacher.py line 392
unlabel_soft(unlabel_examples, predict_result_value, gold_result_tagging, input_mask_tagging, os.path.join(args.output_dir, "soft.tsv"), temp, all_d_set, do_temp=False if args.no_temp else True)
```
- 学生模型训练阶段re-weighting：
```python
# teacher.py line 107
elif args.teacher_score:
  judge_score = torch.zeros(logits_tagging.size(), requires_grad=False).to(self.device)
  for s in range(len(label_ids)):
      judge_score[s] = label_ids[s]
  # logits_tagging = logits_tagging * torch.exp(judge_score)
  
  teacher_score = torch.zeros(logits_tagging.size(), requires_grad=False).to(self.device)
  for s in range(label_soft_ids.size(0)):
      for d in range(label_soft_ids.size(1)):
          teacher_score[s][d] = label_soft_ids[s][d]
  # logits_tagging = logits_tagging * (teacher_score)
  
  logits_tagging = logits_tagging * (judge_score  + teacher_score) / 2
```
